bundles/org.openhab.automation.jrubyscripting/src/main/java/org/openhab/automation/jrubyscripting/internal/JRubyConsoleCommandExtension.java

```diff
@@ -0,0 +1,548 @@
+                            .forEach(file -> console.println("  " + file + (file.isDirectory() ? "/" : "")));
+
+                    // Prevent readLine() from logging a warning
+                    // see:
+                    // https://github.com/apache/karaf/blob/ad427cd12543dc78e095bbaa4608d7ca3d5ea4d8/shell/core/src/main/java/org/apache/karaf/shell/impl/console/ConsoleSessionImpl.java#L549
+                    // https://github.com/jline/jline3/blob/ee4886bf24f40288a4044f9b4b74917b58103e49/reader/src/main/java/org/jline/reader/LineReaderBuilder.java#L90
+                    String previousSetting = System.setProperty("org.jline.reader.support.parsedline", "true");
+                    try {
+                        jrubyConsole.getSession().readLine("\nPress Enter to delete them or Ctrl+C to cancel.", null);
+                        console.println("");
+                    } catch (RuntimeException e) {
+                        // Ctrl+C was pressed
+                        // We can't use a more specific exception type without adding org.jline as bundle dependency
+                        console.println("Operation cancelled.");
+                        return;
+                    } finally {
+                        if (previousSetting != null) {
+                            System.setProperty("org.jline.reader.support.parsedline", previousSetting);
+                        } else {
+                            System.clearProperty("org.jline.reader.support.parsedline");
+                        }
+                    }
+                }
+
+                while (filesToDelete.hasNext()) {
+                    File file = filesToDelete.next();
+                    console.println("Deleting: " + file);
+                    file.delete();
+                    logger.info("Deleted: {}", file);
+                }
+            } catch (IOException e) {
+                console.println("Error: " + e.getMessage());
+                return;
+            }
+        }
+    }
+
+    private synchronized ScriptEngine getEngine(Console console) {
+        ScriptEngine engine;
+
+        ScheduledCompletableFuture<?> unloader = this.engineUnloader;
+        if (unloader != null) {
+            unloader.cancel(false);
+        }
+        if (this.scriptEngine.isEmpty()) {
+            String loadingMessage = "Creating JRuby console script engine...";
+            console.print(loadingMessage);
+            final String scriptIdentifier = "jruby-console-" + UUID.randomUUID().toString();
+            engine = manager.createScriptEngine(scriptType, scriptIdentifier).getScriptEngine();
+            console.print("\r" + " ".repeat(loadingMessage.length()) + "\r");
+            this.scriptEngine = Optional.of(new IdentifierAndEngine(scriptIdentifier, engine));
+        } else {
+            engine = this.scriptEngine.get().engine();
+        }
+        this.engineUnloader = scheduler.schedule(() -> unloadEngine(), Instant.now().plusSeconds(900));
```

Comment:
I see no provision to keep the engine alive if you're continually interacting with the shell

Reply:
With new engine being created for each console/shell session, this is no longer a concern.